source/enemy.js

```diff
@@ -9,13 +9,18 @@ export class Enemy {
     scene;
     killGold;
     name = '';
+
     constructor(data){
         // Add enemy
         this.enemy = data.scene.add.image(data.x, data.y, data.name);
         this.enemy.setOrigin(.5,0).setDepth(2).setScale(.4);
         this.enemy.setInteractive();
+        this.enemy.visible = false;
         this.enemy.on("pointerup", ()=>{
-            this.clickEnemy();
+            // Check if it is the current enemy
+            if (this.enemy.visible) {
+                this.clickEnemy();
+            }
```

Comment:
For some reason using visibility for this rubs me the wrong way, but I can't think of a good reason why it wouldn't be fine, so it should be fine.

Reply:
I had a variable in there specifically for it but this ensures that it can never desync visuals and function